docs/testing.md

```diff
@@ -0,0 +1,115 @@
+# Testing Taps & Targets
+
+The Meltano SDK includes suites of standard tests for both Taps and Targets to help you get started.
+These suites cover most common cases out-of-the-box, and tests are added to the standard suites as new errors are encountered by users in their deployments.
+
+## Test Framework
+
+The Meltano SDK test framework consists of 4 main components:
+
+1. A runner class (`TapTestRunner` and `TargetTestRunner`), responsible for executing Taps/Targets and capturing their output.
+1. A suite dataclass, containing a list of tests.
+1. A test template classes (`TapTestTemplate`, `StreamTestTemplate`, `AttributeTestTemplate` and `TargetTestTemplate`), with methods to `.setup()`, `.test()`, `.validate()` and `.teardown()` (called in that order using `.run()`).
+1. A `get_test_class` factory method, to take a runner and a list of suits and return a `pytest` test class.
+
+## Example Usage
+
+If you created your Tap/Target using the provided cookiecutter templates, you will find the following snippets in `<library_name>/tests/test_core.py`.
+
+### Testing Taps
+
+```python
+import datetime
+
+from singer_sdk.testing import (
+    TapTestRunner,
+    get_test_class,
+    pytest_generate_tests # pytest hook function, required for standard tests
```

Comment:
Is there a way for developers to not have to import this unused function? It probably has to be present in the test namespace for pytest to pick it up, so I wonder if there's a way to register it when installing the SDK itself as a pytest plugin.

Reply:
Thanks for this suggestion @edgarrmondragon - adding `singer_sdk.testing.pytest_plugin` was super easy 👍